lib/ui/screens/group_screen.dart

```diff
@@ -10,48 +10,49 @@ import 'package:reaxit/ui/widgets.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 class GroupScreen extends StatelessWidget {
-  final int pk;
   final ListGroup? group;
 
-  const GroupScreen({super.key, required this.pk, this.group});
+  // By PK
+  final int? pk;
 
-  @override
-  Widget build(BuildContext context) {
-    return BlocProvider<GroupCubit>(
-      create: (context) => GroupCubit(
-        RepositoryProvider.of<ApiRepository>(context),
-        pk: pk,
-      )..load(),
-      child: BlocBuilder<GroupCubit, GroupState>(
-        builder: (context, state) => _Page(
-            state: state,
-            cubit: BlocProvider.of<GroupCubit>(context),
-            listGroup: group),
-      ),
-    );
-  }
-}
+  // By slug
+  final MemberGroupType? groupType;
+  final String? slug;
 
-class BoardScreen extends StatelessWidget {
-  final int since;
-  final int until;
-  final ListGroup? group;
+  // Default: by PK
+  const GroupScreen({
+    super.key,
+    this.group,
+    required this.pk,
+  })  : groupType = null,
+        slug = null;
 
-  const BoardScreen(
-      {super.key, required this.since, required this.until, this.group});
+  // Alternative: by Slug
+  const GroupScreen.bySlug(
+      {super.key, this.group, required this.groupType, required this.slug})
+      : pk = null;
 
   @override
   Widget build(BuildContext context) {
-    return BlocProvider<BoardCubit>(
-      create: (context) => BoardCubit(
-          RepositoryProvider.of<ApiRepository>(context),
-          since: since,
-          until: until)
-        ..load(),
-      child: BlocBuilder<BoardCubit, GroupState>(
+    return BlocProvider<GroupCubit>(
+      create: (BuildContext context) {
+        if (pk != null) {
+          return GroupCubit(
+            RepositoryProvider.of<ApiRepository>(context),
+            pk: pk!,
+          )..load();
+        } else {
+          return GroupCubit.bySlug(
+            RepositoryProvider.of<ApiRepository>(context),
+            groupType: groupType!,
+            slug: slug!,
+          )..load();
+        }
+      },
```

Comment:
This could be a seperate function for readability, like `_selectCubit`

Reply:
Donezo!